common/src/main/java/org/apache/falcon/workflow/engine/AbstractWorkflowEngine.java

```diff
@@ -111,7 +114,8 @@ public abstract InstancesResult getInstanceParams(Entity entity, Date start, Dat
 
     public abstract boolean isNotificationEnabled(String cluster, String jobID) throws FalconException;
 
-    public abstract Boolean isWorkflowKilledByUser(String cluster, String jobId) throws FalconException;
+    public abstract Boolean isWorkflowKilledByUser(String cluster, String jobId,
+                                                   String parentId) throws FalconException;
```

Comment:
Adding a parent Id here makes some assumptions on the hierarchy (coord->wf) which does not hold for native workflow engine. Instead, do a WorkflowInfo to retrieve the parent inside the method.